File: common/src/main/java/com/ssblur/scriptor/recipe/ScriptorRecipes.java
package com.ssblur.scriptor.recipe;

import com.ssblur.scriptor.ScriptorMod;
import dev.architectury.registry.registries.DeferredRegister;
import dev.architectury.registry.registries.RegistrySupplier;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.item.crafting.RecipeSerializer;
import net.minecraft.world.item.crafting.SimpleCraftingRecipeSerializer;

public class ScriptorRecipes {
  public static final DeferredRegister<RecipeSerializer<?>> RECIPES = DeferredRegister.create(ScriptorMod.MOD_ID, Registries.RECIPE_SERIALIZER);

  public static final RegistrySupplier<RecipeSerializer<?>> SPELLBOOK_CLONING = RECIPES.register("spellbook_cloning",
    () -> new SimpleCraftingRecipeSerializer<>(SpellbookCloningRecipe::new));

  public static final RegistrySupplier<RecipeSerializer<?>> SPELLBOOK = RECIPES.register("spellbook",
    SpellbookRecipe.Serializer::new);

  public static final RegistrySupplier<RecipeSerializer<?>> SPELLBOOK_DYEING = RECIPES.register("spellbook_dyeing",
    SpellbookDyeingRecipe.Serializer::new);

  public static void register() { RECIPES.register(); }
}

File: common/src/main/kotlin/com/ssblur/scriptor/recipe/ScriptorRecipes.kt
package com.ssblur.scriptor.recipe

import com.ssblur.scriptor.ScriptorMod
import net.minecraft.world.item.crafting.SimpleCraftingRecipeSerializer

object ScriptorRecipes {
  val SPELLBOOK_CLONING = ScriptorMod.registerRecipeSerializer("spellbook_cloning") {
    SimpleCraftingRecipeSerializer { category -> SpellbookCloningRecipe(category) }
  }
  val SPELLBOOK = ScriptorMod.registerRecipeSerializer("spellbook") { SpellbookRecipe.Serializer() }
  val SPELLBOOK_DYEING = ScriptorMod.registerRecipeSerializer("spellbook_dyeing") { SpellbookDyeingRecipe.Serializer() }

  fun register() {}
}
